src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -1,2180 +1,2682 @@
-                    parentHandle = typeDefinitionMapping[method.ContainingType.ToString()].MethodDefinitions[method][0];
-                }
-                else if (node is IPropertySymbol property)
-                {
-                    parentHandle = typeDefinitionMapping[property.ContainingType.ToString()].PropertyDefinitions[property];
-                }
-                else if (node is IEventSymbol @event)
-                {
-                    parentHandle = typeDefinitionMapping[@event.ContainingType.ToString()].EventDefinitions[@event];
-                }
-                else
-                {
-                    Logger.Log("node not recognized " + node.Kind + " name: " + node.Name);
-                }
-
-                AddCustomAttributes(node.GetAttributes(), parentHandle);
-            }
-        }
-
-        public bool IsPublicNode(MemberDeclarationSyntax node)
-        {
-            return node.Modifiers.Any(modifier => modifier.ValueText == "public") ||
-                (node.Parent is InterfaceDeclarationSyntax && node.Modifiers.Count == 0);
-        }
-
-        public bool IsPrivateNode(AccessorDeclarationSyntax node)
-        {
-            return node.Modifiers.Any(modifier => modifier.ValueText == "private");
-        }
-
-        public bool IsStaticNode(MemberDeclarationSyntax node)
-        {
-            return node.Modifiers.Any(modifier => modifier.ValueText == "static");
-        }
-
-        public bool IsSealedNode(MemberDeclarationSyntax node)
-        {
-            return node.Modifiers.Any(modifier => modifier.ValueText == "sealed");
-        }
-
-        public string GetNamespace()
-        {
-            return string.Join(".", namespaces);
-        }
-
-        public string QualifiedName(string identifier)
-        {
-            return QualifiedName(GetNamespace(), identifier);
-        }
-
-        public string QualifiedName(string @namespace, string identifier)
-        {
-            return string.Join(".", @namespace, identifier);
-        }
-
-        public override void VisitNamespaceDeclaration(NamespaceDeclarationSyntax node)
-        {
-            namespaces.Push(node.Name.ToString());
-            base.VisitNamespaceDeclaration(node);
-            namespaces.Pop();
-        }
-    }
-}
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Generator
+{
+    class Parameter
+    {
+        public Symbol Type;
+        public string Name;
+        public ParameterAttributes Attributes;
+        public bool ByRef;
+
+        public Parameter(Symbol type, string name, ParameterAttributes attributes)
+            :this(type, name, attributes, attributes == ParameterAttributes.Out)

```

Comment:
what is this? a constraint on the parameters?

Reply:
This is just calling the other constructor (public Parameter(Symbol type, string name, ParameterAttributes attributes, bool byRef))